### PR TITLE
Reduce memory consumption in conversion job

### PR DESCRIPTION
**Pull request by mckinsel**
Previously, the conversion jobs were using an enormous amount of memory when converting all the data in the HCA at once. This reduces that to something that runs on an m4.xlarge and removes the relationship between request size and memory usage.

### CODE REVIEW

**Comment by calvinnhieu** [comment]
No changes required to `_to_csv` or `_to_mtx`? Also, bump the docker image version by:
- `make release` in `docker-images/matrix-converter`
- update batch job defn version in `converter_batch.tf`

**Comment by mckinsel** [comment]
No those are coming, so this isn't ready to merge yet. Will bump the version.

**Comment by parthshahva** [comment]
Is 16 GiB (m4.xlarge) the max memory used by conversion jobs or is it a lower amount? We should update https://github.com/HumanCellAtlas/matrix-service/blob/master/terraform/modules/matrix-service/infra/converter_batch.tf#L226 to reflect the new memory and cpu requirements in this PR

**Comment by parthshahva** [comment]
See comments above. This PR still needs a new conversion image version and updates to the terraform config for how much cpu/memory is required. I'm honestly also surprised by the lack of additional test coverage and how large some of these functions have become. It's becoming quite difficult to understand matrix conversions

**Comment by mckinsel** [comment]
I added a functional test that looks at the outputs for each of the three formats. It's too slow to run on every commit, but it compares each file against the redshift outputs directly.

File: tests/functional/test_conversions.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+import loompy
+import s3fs
+import scipy
+
+from matrix.docker.matrix_converter import MatrixConverter
+
+# This is the 2544 pancreas data in a public bucket.
+CELL_MANIFEST = "s3://hca-matrix-conversion-test-data/pancreas/cell_metadata_manifest"
+GENE_MANIFEST = "s3://hca-matrix-conversion-test-data/pancreas/gene_metadata_manifest"
+EXPRESSION_MANIFEST = "s3://hca-matrix-conversion-test-data/pancreas/expression_manifest"
```
Comment (calvinnhieu) [inline tests/functional/test_conversions.py]: Could use `dcp-matrix-test-data` bucket

**Comment by calvinnhieu** [comment]
How long does the test take?

**Pull request merged**